python-checks/src/test/resources/checks/confusingTypeChecking/sillyEquality.py

```diff
@@ -24,3 +24,11 @@ def builtins(x: int, y: str):
 def foo(a: T):
   if a == 42: # FN
     ...
+
+
+from unittest.mock import Mock, MagicMock
+# We should not raise any issues on mocks as they could be monkey patched to fit the comparison type
+def mocks(mock: Mock, magic_mock:MagicMock):
```

Comment:
_Nitpick:_ there are some formatting issues with the type annotation (missing whitespace), feel free to ignore.